websubhub-compiler-plugin-test/src/test/java/io/ballerina/stdlib/websubhub/CompilerPluginTest.java

```diff
@@ -0,0 +1,247 @@
+import io.ballerina.projects.environment.EnvironmentBuilder;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.MessageFormat;
+
+/**
+ * This class includes tests for Ballerina WebSub compiler plugin.
+ */
+public class CompilerPluginTest {
+    private static final Path RESOURCE_DIRECTORY = Paths
+            .get("src", "test", "resources", "ballerina_sources").toAbsolutePath();
+    private static final PrintStream OUT = System.out;
+    private static final Path DISTRIBUTION_PATH = Paths
+            .get("build", "target", "ballerina-distribution").toAbsolutePath();
+
+    @Test
+    public void testValidServiceDeclaration() {
+        Package currentPackage = loadPackage("sample_1");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.diagnostics().size(), 0);
+    }
+
+    @Test
+    public void testCompilerPluginForRemoteMethodValidation() {
+        Package currentPackage = loadPackage("sample_2");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.diagnostics().size(), 1);
+        Diagnostic diagnostic = (Diagnostic) diagnosticResult.diagnostics().toArray()[0];
+        DiagnosticInfo diagnosticInfo = diagnostic.diagnosticInfo();
+        Assert.assertNotNull(diagnosticInfo, "DiagnosticInfo is null for erroneous service definition");
+        Assert.assertEquals(diagnosticInfo.code(), "WEBSUBHUB_102");
+        String expectedMessage = "websubhub:Service should only implement remote methods";
+        Assert.assertEquals(diagnostic.message(), expectedMessage);
+    }
+
+    @Test
+    public void testCompilerPluginForRequiredMethodValidation() {
+        Package currentPackage = loadPackage("sample_3");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.diagnostics().size(), 1);
+        Diagnostic diagnostic = (Diagnostic) diagnosticResult.diagnostics().toArray()[0];
+        DiagnosticInfo diagnosticInfo = diagnostic.diagnosticInfo();
+        Assert.assertNotNull(diagnosticInfo, "DiagnosticInfo is null for erroneous service definition");
+        Assert.assertEquals(diagnosticInfo.code(), "WEBSUBHUB_103");
+        String requiredMethods = String.format("%s,%s,%s,%s,%s,%s,%s",
+                "onRegisterTopic", "onDeregisterTopic", "onUpdateMessage", "onSubscription",
+                "onSubscriptionIntentVerified", "onUnsubscription", "onUnsubscriptionIntentVerified");
```

Comment:
`onSubscription` and `onUnsubscription` are not mandatory right ?

Reply:
* Yes they are optional.